crypto/asn1/charmap.pl

```diff
@@ -82,12 +82,14 @@
 
 # Now generate the C code
 
+# Output year depends on the year of the script.
+my $YEAR = [localtime([stat($0)]->[9])]->[5] + 1900;
```

Comment:
Uhmmmmm, no.  The *output* file might change even if this script file doesn't, and it's the *output* file that we need to update the year on, based on how *it* changes compared to previously.  So basing it on `stat($0)` isn't quite right...

Reply:
I don't understand.  The output is a function of the script and any input files.  If any of those change in year N then the output copyright should be a range ending in year N.

What am I missing?

Reply:
Actually, you're completely right...  D'oh!